akka-javasdk/src/main/scala/akka/javasdk/impl/consumer/ConsumerImpl.scala

```diff
@@ -175,4 +188,12 @@ private[impl] final class MessageContextImpl(
   }
 
   override def tracing(): Tracing = new SpanTracingImpl(span, tracerFactory)
+
+  override def hasLocalOrigin: Boolean = {
+    // empty means we are consuming from a broker
+    // for non-empty origins, it needs to match selfRegion
+    // in dev-mode, both will be "" and therefore always considered as local
+    originRegion.contains(selfRegion)
```

Comment:
Shouldn't we expose `originRegion` and `selfRegion` so that a user can do some custom logic based on that info?

Reply:
Yep, I'm adding it as well